python/device/device.py

```diff
@@ -583,19 +575,20 @@ def send_message():
             self.metrics.send_message_count_not_received_by_service_app.increment()
 
             send_time = time.time()
-            running_op.event.wait()
-
-            logger.info("Telemetry op {} arrived at service".format(running_op.id))
-            self.metrics.send_message_count_not_received_by_service_app.decrement()
-
-            with self.reporter_lock:
-                self.reporter.set_metrics_from_dict(
-                    {
-                        Metrics.LATENCY_QUEUE_MESSAGE_TO_SEND: (send_time - queue_time) * 1000,
-                        Metrics.LATENCY_SEND_MESSAGE_TO_SERVICE_ACK: time.time() - send_time,
-                    }
-                )
-                self.reporter.record()
+            if running_op.event.wait(timeout=self.config[Settings.OPERATION_TIMEOUT_IN_SECONDS]):
+                logger.info("Telemetry op {} arrived at service".format(running_op.id))
+                self.metrics.send_message_count_not_received_by_service_app.decrement()
+
+                with self.reporter_lock:
+                    self.reporter.set_metrics_from_dict(
+                        {
+                            Metrics.LATENCY_QUEUE_MESSAGE_TO_SEND: (send_time - queue_time) * 1000,
+                            Metrics.LATENCY_SEND_MESSAGE_TO_SERVICE_ACK: time.time() - send_time,
+                        }
+                    )
+                    self.reporter.record()
+            else:
```

Comment:
I think incrementing a counter here and using that counter value in `check_failure_counts()` would be more consistent rather than using `send_message_count_not_received_by_service_app`

Reply:
I think you're right.  I'm going to make this consistent with other ops in my next PR.